Tests/FluentAssertions.Shared.Specs/ObjectAssertionSpecs.cs

```diff
@@ -454,6 +454,68 @@ public void When_object_is_assignable_to_the_expected_type_it_should_cast_the_re
             act.ShouldThrow<AssertFailedException>().WithMessage("*Expected*Other*Actual*");
         }
 
+        [TestMethod]
+        public void When_asserting_an_object_is_assignable_its_own_type_using_type_instance_it_should_succeed()
```

Comment:
Don't repeat the `When_asserting_an_object_is_assignable`. I'd rather put all the methods in a region and keep the names short and clear.